lib/private/Group/Database.php

```diff
@@ -331,6 +335,42 @@ public function groupExists($gid) {
 		return false;
 	}
 
+	/**
+	 * {@inheritdoc}
+	 */
+	public function groupsExists(array $gids): array {
+		$notFounsGids = [];
+		$existingGroups = [];
+
+		// In case the data is already locally accessible, not need to do SQL query
+		// or do a SQL query but with a smaller in clause
+		foreach ($gids as $gid) {
+			if (isset($this->groupCache[$gid])) {
+				$existingGroups[] = $gid;
+			} else {
+				$notFounsGids[] = $gid;
```

Comment:
```suggestion
				$notFoundGids[] = $gid;
```
There are some of them :)